src/miner.cpp

```diff
@@ -1049,7 +1049,7 @@ void StakeMiner(CWallet *pwallet)
                 StakeBlock.nVersion = 8;
             if(IsV9Enabled(pindexPrev->nHeight+1))
                 StakeBlock.nVersion = 9;
-            if(pindexPrev->nVersion == 10)
+            if(IsV10Enabled(pindexPrev->nHeight + 1))
                 StakeBlock.nVersion = 10;
```

Comment:
It might be worth to remove the other alternatives. Wont be mining bv8 blocks anymore.